routing/datasource.go

```diff
@@ -25,6 +39,10 @@ type incomingData struct {
 	deletedIds     []string
 }
 
+// continously receives route definitions from a data client on the the output channel.
+// The function does not return. When started, it request for the whole current set of
+// routes, and continues polling for the subsequent updates. When a communication error
+// occurs, it re-requests the whole valid set, and continues polling.
```

Comment:
Should we add specific information about keeping the currently know status/content?